pkgs/development/libraries/vulkan-loader/default.nix

```diff
@@ -17,17 +17,23 @@ stdenv.mkDerivation rec {
     sha256 = "0zhrwj1gi90x2w8gaaaw5h4b969a8gfy244kn0drrplhhb1nqz3b";
   };
 
-  nativeBuildInputs = [ pkgconfig ];
+  nativeBuildInputs = [ pkgconfig addOpenGLRunpath ];
   buildInputs = [ cmake python3 xlibsWrapper libxcb libXrandr libXext wayland ];
   enableParallelBuilding = true;
 
   cmakeFlags = [
-    "-DFALLBACK_DATA_DIRS=${libGL_driver.driverLink}/share:/usr/local/share:/usr/share"
+    "-DFALLBACK_DATA_DIRS=${addOpenGLRunpath.driverLink}/share:/usr/local/share:/usr/share"
     "-DVULKAN_HEADERS_INSTALL_DIR=${vulkan-headers}"
   ];
 
   outputs = [ "out" "dev" ];
 
+  # Set RUNPATH so that driver libraries in /run/opengl-driver(-32)/lib can be found.
+  # See the explanation in addOpenGLRunpath.
+  postFixup = ''
+    addOpenGLRunpath $out/lib/libvulkan.so
```

Comment:
This _should not_ be needed.

Vulkan Loader doesn't need the `openGLRunpath` hardcoded because it will look  up libraries not in the opengl path but in whatever   the ICD json points to.  So vulkan will never look in `/run/opengl-driver/lib`,
it will instead look in `/nix/store/y99l7729c0157a52y9qfzi3zg4f94b57-mesa-noglu-18.3.4-drivers/lib/libvulkan_intel.so`     because the `ICD` in `/run/opengl-driver/share/vulkan/icd.d` says so

```
cat /run/opengl-driver/share/vulkan/icd.d/intel_icd.x86_64.json 
{
    "ICD": {
        "api_version": "1.1.90",
        "library_path": "/nix/store/y99l7729c0157a52y9qfzi3zg4f94b57-mesa-noglu-18.3.4-drivers/lib/libvulkan_intel.so"
    },
    "file_format_version": "1.0.0"
}

```

Reply:
It is currently needed for NVidia because the library_path is only a name:
```
$ cat /run/opengl-driver-32/share/vulkan/icd.d/nvidia.json 
{
    "file_format_version" : "1.0.0",
    "ICD": {
        "library_path": "libGLX_nvidia.so",
        "api_version" : "1.1.99"
    }
}
```
I think we should patch it and then remove the runpath.

Reply:
I agree. any other drivers that would need similar fixups? ;-)

Reply:
Opened #62870 to set absolute paths in NVidia ICD files.
Also related:
- #62859: Fix that will enable actually dropping RUNPATH in dispatch libraries and LD_LIBRARY_PATH in `opengl.nix`.
- #62869: Will enable also dropping `XDG_DATA_DIRS` in `opengl.nix`.